array_equals.rb

```diff
@@ -0,0 +1,21 @@
+# Determines if the two input arrays have the same count of elements
+# and the same integer values in the same exact order
+def array_equals(array1, array2)
+  if array1 == nil && array2 == nil
+    return true
+  elsif array1 == nil && array2 != nil
+    return false
+  elsif array1 != nil && array2 == nil
+    return false
+  end
+  if array1.length != array2.length
+    return false
+  end
+  n = array1.length
+  (0..n - 1).each do |i|
```

Comment:
I think this could be simplified by just doing a times loop (array1.length.times do).  That way you would not have to define n and then use n in the each do block.

Reply:
Great point! I applied your suggestion and it works wonder! Yay